enterprise/backend/test/metabase_enterprise/serialization/api_test.clj

```diff
@@ -205,3 +207,11 @@
         ;; in `api/on-response!`, so maybe add some Thread/sleep here.
         (is (= known-files
                (set (.list (io/file api.serialization/parent-dir)))))))))
+
+(deftest find-serialization-dir-test
+  (testing "We are able to find serialization dir even in presence of various hidden dirs"
+    (let [dst (io/file api.serialization/parent-dir (u.random/random-name))]
```

Comment:
We probably want to use a temp file and mark it for clean up, or does something else take care of that?

Reply:
uhhh I got so used to serialization api cleaning up its stuff that did not do anything here, but obviously it needs to be cleaned up. Don't want to use Java's temp stuff APIs, they are too awkward, though.

Reply:
Sneaky solution, I kinda like it.

Reply:
It's the one [used in actual code](https://github.com/metabase/metabase/blob/serdes-better-dir-detect/enterprise/backend/src/metabase_enterprise/serialization/api.clj#L97) 😁